### DISCUSSION TITLE
FLAML DOI?

### DISCUSSION THREAD

**Discussion by ZviBaratz** [microsoft/FLAML#617] [Ideas]
Hi,

I was wondering what's the general feeling regarding using something like [Zenodo](https://zenodo.org/) to generate a DOI for the code itself so that it can be cited easily. I was just working on the relevant methods section for my research and of course included the citation included in the docs' [research](https://microsoft.github.io/FLAML/docs/Research) page, but I thought it could be a great addition.

**Comment by qingyun-wu** [comment]
Hi @[ZviBaratz](https://github.com/ZviBaratz),

Thank you for raising this discussion. I am not familiar with this type of citation format but seems. Do you have an example of it?

**Reply by ZviBaratz** [reply]
Sure, it is quite common in neuroimaging packages, but I think it is becoming quite popular for scientific packages in general. A couple of examples for widely used libraries that use it are [nibabel](https://github.com/nipy/nibabel), [nipype](https://github.com/nipy/nipype), and [mriqc](https://github.com/nipreps/mriqc), there are many more.

**Reply by qingyun-wu** [reply]
Thank you very much @[ZviBaratz](https://github.com/ZviBaratz) for sharing the examples. 

Do you know the additional benefits of using a DOI for the code itself when a citation to the research paper associated with the code is available?

And I notice that [mriqc](https://github.com/nipreps/mriqc) does mention the following instruction in the citation section of their README file:
"When using MRIQC, please include the following citation:
Esteban O, Birman D, Schaer M, Koyejo OO, Poldrack RA, Gorgolewski KJ; MRIQC: Advancing the Automatic Prediction of Image Quality in MRI from Unseen Sites; PLOS ONE 12(9):e0184661; doi:[10.1371/journal.pone.0184661](https://doi.org/10.1371/journal.pone.0184661)."

I am worried that having two ways to cite the lib may confuse some people. But if the additional benefits outweigh the downside, we can consider generating the DOI. 

Please chime in if you have any thoughts/suggestions about it. @[sonichi](https://github.com/sonichi), @liususan091219 

Thank you!

**Reply by ZviBaratz** [reply]
If I understand correctly, the two methods do not contradict each other. The article is meant to provide a reference for the context, motivation, general idea of the project, important implementation details, etc., whereas the Zenodo DOI is meant to provide a reference to a particular version of the code itself, with the main advantage of improving the reproducibility of derived work.

**Reply by sonichi** [reply]
I'm not familiar with Zenodo DOI. For the benefit of improving the reproducibility of derived work, is it not sufficient to specify the version number of the library used? For example, flaml v1.0.9?

**Reply by ZviBaratz** [reply]
I believe in terms of individuals trying to reproduce particular work it should be the same, but using the Zenodo DOI provides a standardized and easily traceable method to query and follow citations across papers, index and catalog them, etc. I think it would be better to rely on external sources to advocate the advantages, e.g. [this GenR article](https://genr.eu/wp/cite/), [this UC Berkeley library  guide](https://guides.lib.berkeley.edu/citeyourcode), [here](https://coderefinery.github.io/github-without-command-line/doi/), etc. GitHub docs actually also [briefly mention this](https://docs.github.com/en/repositories/archiving-a-github-repository/referencing-and-citing-content) as well, though just as a reference. In any case, no agenda or particular interest here, just a suggestion.
